pkg/shares/counter.go

```diff
@@ -0,0 +1,82 @@
+package shares
+
+import (
+	"github.com/celestiaorg/celestia-app/pkg/appconsts"
+)
+
+type CompactShareCounter struct {
+	lastShares    int
+	lastRemainder int
+	shares        int
+	remainder     int
+}
+
+// NewCompactShareCounter creates a new instance of a counter which calculates the amount
+// of compact shares a set of data will be split into.
+func NewCompactShareCounter() *CompactShareCounter {
+	return &CompactShareCounter{}
+}
+
+// Add adds the length of the data to the counter and returns the amount of shares
+// the counter has been increased by.
+func (c *CompactShareCounter) Add(dataLen int) int {
+	// Increment the data len by the varint that will prefix the data.
+	dataLen += DelimLen(uint64(dataLen))
+
+	// save a copy of the previous state
+	c.lastRemainder = c.remainder
+	c.lastShares = c.shares
+
+	// if this is the first share, calculate how much is taken up by dataLen
+	if c.shares == 0 {
+		if dataLen+c.remainder > appconsts.FirstCompactShareContentSize {
```

Comment:
[blocking] I'm having a tough time understanding why this isn't

```suggestion
		if dataLen >= appconsts.FirstCompactShareContentSize-c.remainder {
```

can you please explain the rationale? I think the suggestion matches line 43 which intuitively makes more sense to me

Reply:
Yes I think your rationale makes sense (to change from `>` to `>=`) because when it equals a new share you just want to increment the shares. I caught myself at the one below but didn't change it here. Good catch